wsr-tic-tac-toe/src/redux/reducers/index.ts

```diff
@@ -0,0 +1,137 @@
+import {
+    WinningScenarioActionTypes,
+    BoardState,
+    BoardStateActionTypes,
+    SquareState,
+    GAME_STATE,
+} from '../types';
+import {
+    CLEAR_WINNING_SCENARIOS,
+    CREATE_BOARD,
+    REMOVE_BOARD,
+    GENERATE_WINNING_SCENARIOS,
+    SELECT_SQUARE,
+    CLEAR_BOARD,
+    SET_GAME_STATE,
+} from '../actions';
+import { combineReducers } from 'redux';
+const initialWinningScenariosState: Array<Array<number>> = [];
+
+export function winningScenariosReducer(
+    state = initialWinningScenariosState,
+    action: WinningScenarioActionTypes
+) {
+    switch (action.type) {
+        case CLEAR_WINNING_SCENARIOS: {
+            return initialWinningScenariosState; // return initialWinningScenariosState
+        }
+        case GENERATE_WINNING_SCENARIOS: {
+            let boardSize = action.payload;
+            let WSh: Array<Set<number>> = []; // array of horizontal winning sets
+            let WSv: Array<Set<number>> = []; // array of vertical winning sets
+            for (let i: number = 0; i < boardSize; i++) {
+                let tempSet = new Set<number>();
+                for (let j: number = 0; j < boardSize; j++) {
+                    tempSet.add(boardSize * i + j);
+                }
+                WSh = [...WSh, tempSet];
+            }
+
+            for (let i: number = 0; i < boardSize; i++) {
+                let tempSet = new Set<number>();
+                for (let j: number = 0; j < boardSize; j++) {
+                    tempSet.add(boardSize * j + i);
+                }
+                WSv = [...WSv, tempSet];
+            }
+
+            let diagonalSet1 = new Set();
+            let diagonalSet2 = new Set();
+            for (let i = 0; i < boardSize; i++) {
+                diagonalSet1.add(i * boardSize + i);
+                diagonalSet2.add((i + 1) * (boardSize - 1));
+            }
+            let winningScenarios = [];
+            winningScenarios = [
+                ...WSh.map((set) => Array.from(set)),
+                ...WSv.map((set) => Array.from(set)),
+                Array.from(diagonalSet1),
+                Array.from(diagonalSet2),
+            ];
+            return [...winningScenarios];
+        }
+        default:
+            return state;
+    }
+}
+
+const initialBoardState: BoardState = {
+    boardSize: 0,
+    created: false,
+    squares: [],
+    turnOrder: 'player1',
+    gameState: GAME_STATE.ONGOING
+};
+
+export function boardReducer(state = initialBoardState, action: BoardStateActionTypes) {
+    switch (action.type) {
+        case CREATE_BOARD: {
+            let boardSize = action.payload;
+            let initializedSquares: SquareState[] = [];
+            // let squares: SquareState[][] = [];
+            for (let i: number = 0; i < boardSize * boardSize; i++) {
+                initializedSquares = [
```

Comment:
since you are creating the initializedSquares array just two lines ago, you can .push into it, instead of spreading (no matter how much I like spreading, it's not quite appropriate here).